### PR TITLE
Lineage best practice #863

### PR DESCRIPTION
**Pull request by m-mohr**
**Related Issue(s):** #863

**Proposed Changes:**

1. Moved the derived_from note to best practices, makes the item spec a bit slimmer and bundles details at a single place.
2. Added idea from #863 for lineage information to BP.

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [ ] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

File: best-practices.md [hunk 1]
```diff
@@ -397,3 +398,15 @@ basic geographic filtering from listeners.
...
 database, but it could just as easily be a server-based process.
+
+## Provenance
+
+A full provenance model is far beyond the scope of STAC, and the goal is to align with any good independent spec 
+that comes along for that. The [`derived_from` relation type in links](item-spec/item-spec.md#relation-types) is seen as
+a way to encourage fuller specs and at least start a linking structure that can be used as a jumping off point for more
+experiments in provenance tracking.
+
+Providers can provide [lineage](https://github.com/radiantearth/stac-spec/issues/863) information in the `providers` property,
```
Comment (cholmes) [inline best-practices.md:409]: A little odd that we're linking to an issue here. But I hope the intent is we get to an extension in the spec before too long?

Comment (m-mohr) [reply best-practices.md:409]: The issue has some background on lineage, like that we may add sci:lineage and other ways you could add those info. Wanted to link that somewhere, but maybe not the best place? Happy to adopt any suggestion.

Comment (cholmes) [reply best-practices.md:409]: I'd probably just summarize the key info from that ticket here...

Comment (m-mohr) [reply best-practices.md:409]: But I wouldn't consider all that best practices though...

**Comment by m-mohr** [comment]
We'll have a discussion on that tomorrow, hope to get that in afterwrds.

**Comment by m-mohr** [comment]
I'll try to make a small extension for lineage fields in the provider arrays....

**Comment by m-mohr** [comment]
I think we should merge this with #907!

File: best-practices.md (hunk 1 above)
Comment (cholmes) [reply best-practices.md:409]:
```suggestion
Providers can provide lineage information (see [this issue](https://github.com/radiantearth/stac-spec/issues/863) for background) in the `providers` property,
```

Comment (cholmes) [reply best-practices.md:409]: Ok, the above suggestion would be a bit better for me - as a user I'd expect a link in a best practice doc to link me to some authoritative definition, not to a discussion.

**Comment by cholmes** [comment]
+1 on merging with #907

**Comment by matthewhanson** [comment]
+1 closing in favor of #907

**Comment by m-mohr** [comment]
Closed in favor of #907.

**Pull request closed without merging**